ckanext/short_urls/model.py

```diff
@@ -0,0 +1,50 @@
+import string
+import random
+from sqlalchemy import Column, types, UniqueConstraint
+from sqlalchemy.ext.declarative import declarative_base
+from ckan.model.meta import metadata
+
+Base = declarative_base(metadata=metadata)
+
+OBJECT_TYPE_DATASET = 'dataset'
+OBJECT_TYPE_RESOURCE = 'resource'
+
+
+def _generate_random_string(string_length=6):
+    # taken from https://bit.ly/3nxGSMo
+    alphanumeric_chars = string.ascii_lowercase + string.digits
+    return ''.join(
+        random.SystemRandom().choice(alphanumeric_chars)
+        for _ in range(string_length)
+    )
+
+
+class ShortUrls(Base):
+    """
+    Stores a short_url pointing to a specific dataset or resource
+    """
+    __tablename__ = 'short_url'
+
+    id = Column(types.Integer, primary_key=True, nullable=False)
+    hash = Column(
```

Comment:
`hash` is not a good name for this column. It creates connotations with hash functions which this feature has nothing to do.
Please rename to sth else. Maybe `path` as it'll be used as url path in final URL?